src/api/index.js

```diff
@@ -0,0 +1,272 @@
+            headers: {
+                'Accept': 'application/json',
+                'Content-Type': 'application/json',
+                ...!!token && {Authorization: `Bearer ${token}`},
+
+            },
+            credentials: 'include',
+
+        };
+
+        const mergedData = mergeRecursive(data, _data);
+
+        return fetch(url, mergedData)
+            .then(response => {
+                if (response.status <= 300) {
+                    return response;
+                }
+                console.log(response.status)
+                if (response.status === 302) {
+                    redirect(response.headers.location);
+                }
+
+                return response.json().then(data => {
+                    if (response.status >= 300) {
+                        return data.message
+                    }
+                    return data;
+                });
+            })
+            .catch(e => Error(e));
+
+    };
+
+    fetchCreateUser = user => {
+        return this.makeRequest({
+            url: `${this.url}/signUp`,
+            data: {
+                method: 'POST',
+                body: JSON.stringify({
+                    username: user.username,
+                    password: user.password,
+                    email: user.email
+                })
+            }
+        })
+    };
+
+
+    fetchLoginUser = user => {
+        return this.makeRequest({
+            url: `${this.url}/login`,
+            data: {
+                method: 'POST',
+                body: JSON.stringify({
+                    email: user.email,
+                    password: user.password
+                })
+            }
+        })
+    };
+
+    fetchLoginWithGoogleUser = () => {
+        return this.makeRequest({
+            url: `${this.url}/login/google`,
+            data: {
+                method: 'GET',
+
+            }
+        })
+    };
+
+    fetchGetTokenUser = () => {
+        return this.makeRequest({
+            url: `${this.url}/getToken`,
+            data: {
+                method: 'GET',
+
+            }
+        })
+    };
+
+    fetchArtists = () => {
+        return this.makeRequest({
+            url: `${this.url}/artists`,
+            data: {
+                method: 'GET',
+            },
+            token: localStorage.token
+        })
+    };
+
+    fetchAddToDict = (WordId, image, songId) => {
+        return this.makeRequest({
+            url: `${this.url}/user/dict`,
+            data: {
+                method: 'POST',
+                body: JSON.stringify({
+                    word_id: WordId,
+                    song_id: songId,
+                    prime_picture: image,
+                })
+            },
+            token: localStorage.token
+
+        })
+    };
+
+    fetchDict = () => {
+        return this.makeRequest({
+            url: `${this.url}/user/dict`,
+            data: {
+                method: 'GET'
+            },
+            token: localStorage.token
```

Comment:
what about moving this token field to makeReques method?
it seems that in most methods you use thi field